interfaces/repo.go

```diff
@@ -673,6 +706,34 @@ func (r *Repository) connected(snapName, plugOrSlotName string) ([]ConnRef, erro
 	return conns, nil
 }
 
+func (r *Repository) Connections(snapName string) ([]ConnRef, error) {
```

Comment:
I would have expected []*ConnRef

Reply:
This is in line with other methods in repo that return ConnRef (ResolveDisconnect, Connected).

Reply:
well, we are really not that consistent with xxxRef  types,  I see

NewConnRef(plug *snap.PlugInfo, slot *snap.SlotInfo) *ConnRef
vs
func ParseConnRef(id string) (ConnRef, error)
vs
func (slot *Slot) Ref() SlotRef
vs
func (slot *ConnectedSlot) Ref() *SlotRef

not here, but for a follow up, unless we compare these xxxRef a lot with each other or use them as keys in maps a lot,  I would use  *xxxRef everywhere,  they are big enough (>=128bits) for that to make sense

also seeing *NewConnRef(...) a lot is strange

Reply:
Ok, will do in a followup.

Reply:
Fair enough; I made the change in separate PR where we can discuss the merit.